localstack/services/sns/provider.py

```diff
@@ -1559,3 +1588,81 @@ def unsubscribe_sqs_queue(queue_url):
             sub_url = subscriber.get("sqs_queue_url") or subscriber["Endpoint"]
             if queue_url == sub_url:
                 subscriptions.remove(subscriber)
+
+
+@singleton_factory
+def register_sns_api_resource():
+    """Register the platform endpointmessages retrospection endpoint as an internal LocalStack endpoint."""
+    get_internal_apis().add(
+        PLATFORM_ENDPOINT_MSGS_ENDPOINT, SNSServicePlatformEndpointMessagesApiResource()
+    )
+
+
+def _format_platform_endpoint_messages(sent_messages: List[Dict[str, str]]):
+    """
+    This method format the messages to be more readable and undo the format change that was needed for Moto
+    Should be removed once we refactor SNS.
+    """
+    validated_keys = [
+        "TargetArn",
+        "Message",
+        "Message",
+        "MessageAttributes",
+        "MessageStructure",
+        "Subject",
+    ]
+    formatted_messages = []
+    for sent_message in sent_messages:
+        msg = {
+            key: value[0] if isinstance(value, list) else value
+            for key, value in sent_message.items()
+            if key in validated_keys
+        }
+        formatted_messages.append(msg)
+
+    return formatted_messages
+
+
+class SNSServicePlatformEndpointMessagesApiResource:
+    """Provides a REST API for retrospective access to platform endpoint messages sent via SNS.
+
+    This is registered as a LocalStack internal HTTP resource.
+
+    This endpoint accepts:
+    - GET param `region`: selector for AWS `region`. If not specified, return default "us-east-1"
+    - GET param `endpointArn`: filter for `endpointArn` resource in SNS
+    - DELETE param `region`: will delete saved messages for `region`
+    - DELETE param `endpointArn`: will delete saved messages for `endpointArn`
+    """
+
+    def on_get(self, request: Request):
+        region = request.args.get("region", "us-east-1")
+        filter_endpoint_arn = request.args.get("endpointArn")
+        store: SnsStore = sns_stores[get_aws_account_id()][region]
```

Comment:
since we support multi-account now, maybe we should also allow passing of the account id as query param, and fall back to `get_aws_account_id()` only if necessary? since `get_aws_account_id` requires an auth header to return something helpful.